frontend/packages/core/src/dialog.tsx

```diff
@@ -57,42 +57,31 @@ const DialogActions = styled(MuiDialogActions)({
 
 export interface DialogContentProps {}
 export interface DialogActionsProps {}
+export type DialogCloseReasons = "closeButtonClick" | "escapeKeyDown" | "backdropClick"
 export interface DialogProps extends Pick<MuiDialogProps, "open"> {
   title: string;
   children:
     | React.ReactElement<DialogContentProps>
     | React.ReactElement<DialogContentProps>[]
     | React.ReactElement<DialogActionsProps>
     | React.ReactElement<DialogActionsProps>[];
-  onClose?: () => void;
+  onClose: (event?: object, reason?: DialogCloseReasons) => void;
```

Comment:
is `object` too generic? MuiDialog uses `{}` but our close button would pass a MouseEvent

Reply:
If we can make it more specific that would be great but it's not a blocker